From 0a1b86992d179640a81d11bcdc24d409265ccc18 Mon Sep 17 00:00:00 2001
From: Jeffrey Mouroux <jeff.mouroux@xilinx.com>
Date: Thu, 13 Jul 2017 18:31:13 -0700
Subject: [PATCH 0213/1852] media: xilinx: dma: Enable use of Framebuffer Write
 IP in V4L2 Driver

commit 22eed07aec3d64a646ba09253c5032aee999ad4b from
https://github.com/Xilinx/linux-xlnx.git

The Xilinx Video Framebuffer driver is video 'format aware' and
requires clients to pass video format information as part of DMA
programming. As the DMA Engine interface does not support passing
this information, a configuration object is passed through the
dma channel's private data to the Framebuffer Driver with this
information.

Signed-off-by: Jeffrey Mouroux <jeff.mouroux@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dma.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-dma.c b/drivers/media/platform/xilinx/xilinx-dma.c
index 098c69bfe5c7..6daf8b0c2e82 100644
--- a/drivers/media/platform/xilinx/xilinx-dma.c
+++ b/drivers/media/platform/xilinx/xilinx-dma.c
@@ -10,6 +10,7 @@
  */
 
 #include <linux/dma/xilinx_dma.h>
+#include <linux/dma/xilinx_frmbuf.h>
 #include <linux/lcm.h>
 #include <linux/list.h>
 #include <linux/module.h>
@@ -349,6 +350,8 @@ static void xvip_dma_buffer_queue(struct vb2_buffer *vb)
 	dma_addr_t addr = vb2_dma_contig_plane_dma_addr(vb, 0);
 	u32 flags;
 
+	xilinx_xdma_v4l2_config(dma->dma, dma->format.pixelformat);
+
 	if (dma->queue.type == V4L2_BUF_TYPE_VIDEO_CAPTURE) {
 		flags = DMA_PREP_INTERRUPT | DMA_CTRL_ACK;
 		dma->xt.dir = DMA_DEV_TO_MEM;
-- 
2.31.1

